### PR TITLE
feat(quit): make sure to ask before discarding changes

### LINKED ISSUE #1239: [Bug]: smart quit that doesn't discard unwritten changes
### Problem description

`<leader>c` and `<leader>q` exit without a warning. There may be unsaved changes, thereby work can be lost unexpectedly.

I have the following script in my init.vim that handles buffers and windows in a sane way. I intend to convert it to lua eventually (my lua skills are novice level atm). Here's the script in case anyone else wants to take a crack at it:
```

function! SmartQuit()
    let num_buffers = len(getbufinfo({'buflisted':1}))
    if num_buffers == 1
      try
        execute ":silent quit"
      catch /E37:/
        " Unwritten changes
        echo "E37: Discard changes?  y = yes, n = no, w = write and quit: "
        let ans = nr2char(getchar())
        if ans == "y" 
          execute "quit!"
        elseif  ans == "w" 
          execute "write"
          execute "quit"
        else
          call feedkeys('\<ESC>')
        endif
      endtry
    else
      try
        execute "bd"
      catch /E89:/
        " Unwritten changes
        echo "E89: Discard changes?  y = yes, n = no, w = write and quit: "
        let ans = nr2char(getchar())
        if ans == "y" 
          execute "bd!"
        elseif  ans == "w" 
          execute "write"
          execute "bd"
        else
          call feedkeys('\<ESC>')
        endif
      endtry
    endif
endfunction
```

### LunarVim version

rolling

### Neovim version (>= 0.5)

NVIM v0.6.0-dev+5-g05330ba50

### Relevant log output

_No response_

### Screenshots

_No response_

### I have

- [X] Read the readme
- [X] Read the wiki
- [X] Searched for similar issues

### PR DESCRIPTION
**Pull request by abzcoding**
# Description

Ask for confirmation if there are unsaved changes

Fixes #1239

### CODE REVIEW

File: lua/lvim/utils/functions.lua [hunk 1]
```diff
@@ -0,0 +1,17 @@
+local M = {}
+
+M.smart_quit = function()
```
Comment (kylo252) [inline lua/lvim/utils/functions.lua:3]:
```suggestion
M.function smart_quit()
```
nitpick :')

File: lua/lvim/utils/functions.lua [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  if vim.api.nvim_eval [[&modified]] == 1 then
```
Comment (kylo252) [inline lua/lvim/utils/functions.lua:4]:
```suggestion
  local bufnr = vim.api.nvim_get_current_buf()
  local modified = vim.api.nvim_buf_get_option(bufnr, "modified")
  if modified then
```

I think this should work the same, at least it seems fine in my testing.

File: lua/lvim/utils/functions.lua [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    vim.ui.input({
+      prompt = "You have unsaved changes. Quit anyway? (y/n) ",
+    }, function(input)
+      if input == "y" then
+        vim.cmd "q!"
+      end
+    end)
+  else
+    vim.cmd "q!"
```
Comment (kylo252) [inline lua/lvim/utils/functions.lua:13]:
```suggestion
    vim.cmd "q"
```

I'm not sure if there's another reason why `quit` wouldn't continue, other than a modified file, but maybe not risk it?

File: lua/lvim/utils/functions.lua (hunk 1 above)
Comment (abzcoding) [reply lua/lvim/utils/functions.lua:3]: are you sure that is the correct syntax?
Screen Shot 2022-05-03 at 6 33 38 PM

Comment (abzcoding) [reply lua/lvim/utils/functions.lua:3]: maybe you meant ?
```lua
function M.smart_quit()
```

**Pull request merged**